### PR TITLE
Assignment 3 Submission

### PR DESCRIPTION
**Pull request by Nematodes**
This is my assignment 3 submission. Future commits are unlikely unless issues are brought to my attention.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -344,4 +344,8 @@ bin/
 .DS_Store
 
 # Remove VS Code Configuration Folder
+<<<<<<< HEAD
 .vscode/
+=======
+.vscode/
+>>>>>>> 0a27711f4faa569fd610137756833e742ee7408a
```
Comment (Keboo) [inline .gitignore]: Minor thing here. It looks like you had a merge conflict. You will want to pick either line 348 or line 350 (but they appear equal), and remove the other merge conflict lines. Docs [here](https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/)

File: Assignment3/RockPaperScissors/RockPaperScissorsUnitTester/RockPaperScissorsUnitTester.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            Assert.IsTrue(RockPaperScissors.ValidatePlayerActionInput("paper"));
+            Assert.IsTrue(RockPaperScissors.ValidatePlayerActionInput("scissors"));
+
+            Assert.IsFalse(RockPaperScissors.ValidatePlayerActionInput("Rock"));
+            Assert.IsFalse(RockPaperScissors.ValidatePlayerActionInput(" rock"));
+            Assert.IsFalse(RockPaperScissors.ValidatePlayerActionInput("rock paper"));
+            Assert.IsFalse(RockPaperScissors.ValidatePlayerActionInput("paperscissors"));
+            Assert.IsFalse(RockPaperScissors.ValidatePlayerActionInput("tangerine"));
+            Assert.IsFalse(RockPaperScissors.ValidatePlayerActionInput(""));
+        }
```
Comment (Keboo) [inline Assignment3/RockPaperScissors/RockPaperScissorsUnitTester/RockPaperScissorsUnitTester.cs]: You might also consider testing `null`

File: Assignment3/RockPaperScissors/RockPaperScissors/RockPaperScissors.cs [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                    return GameAction.Paper;
+                default:
+                    return GameAction.Scissors;
+            }
+        }
+
+        // The assignment specifications do state that valid input will always be given...but that would make the program more boring than it already is
+        public static bool ValidatePlayerActionInput(String playerInput)
+        {
+            if (String.Compare(playerInput, "rock") == 0 || String.Compare(playerInput, "paper") == 0 || String.Compare(playerInput, "scissors") == 0)
```
Comment (Keboo) [inline Assignment3/RockPaperScissors/RockPaperScissors/RockPaperScissors.cs]: If you are only testing for equality, it is better to use `string.Equals`
